src/lazy/encoder/writer.rs

```diff
@@ -207,9 +228,10 @@ impl<'value, V: ValueWriter> AnnotatableWriter for ApplicationValueWriter<'value
     where
         Self: 'a,
     {
-        if self.encoding.symbol_creation_policy == SymbolCreationPolicy::WriteProvidedToken {
+        if self.has_inline_symbol_text() {
```

Comment:
🗺️ `has_inline_symbol_text` is more specific than the initial `SymbolCreationPolicy`. It may be that I can remove that concept, but if so I'll do that in another PR.